app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,57 @@
+class TweetsController < ApplicationController
+#==============================tweets================================ 
+  get '/tweets' do 
+    redirect '/login' if !logged_in?
+    
+    erb :"tweets/tweets"
+  end
+#===============================new================================== 
+  get '/tweets/new' do 
+    redirect '/login' if !logged_in?
+    
+    erb :'tweets/create_tweet'
+  end
+#-------------------------------------------------------------------- 
+  post '/tweets' do 
+    redirect '/tweets/new' if empty?(params)
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.

Reply:
Also learned about this after fwitter lol using it here 

https://github.com/AlphaDaniel/snalgos/blob/a7bdc78edceb11f62a7dff6249fd49c8617e4083/app/models/user.rb#L7